src/selective.d

```diff
@@ -1,181 +1,228 @@
-	}
-	
-	// Match against skip_dir, skip_file & sync_list entries
-	bool isPathExcludedMatchAll(string path)
-	{
-		return .isPathExcluded(path, paths) || .isPathMatched(path, mask) || .isPathMatched(path, dirmask);
-	}
-	
-	// is the path a dotfile?
-	bool isDotFile(string path)
-	{
-		// always allow the root
-		if (path == ".") return false;
-
-		path = buildNormalizedPath(path);
-		auto paths = pathSplitter(path);
-		foreach(base; paths) {
-			if (startsWith(base, ".")){
-				return true;
-			}
-		}
-		return false;
-	}
-}
-
-// test if the given path is not included in the allowed paths
-// if there are no allowed paths always return false
-private bool isPathExcluded(string path, string[] allowedPaths)
-{
-	// always allow the root
-	if (path == ".") return false;
-	// if there are no allowed paths always return false
-	if (allowedPaths.empty) return false;
-
-	path = buildNormalizedPath(path);
-	foreach (allowed; allowedPaths) {
-		auto comm = commonPrefix(path, allowed);
-		if (comm.length == path.length) {
-			// the given path is contained in an allowed path
-			return false;
-		}
-		if (comm.length == allowed.length && path[comm.length] == '/') {
-			// the given path is a subitem of an allowed path
-			return false;
-		}
-	}
-	return true;
-}
-
-// test if the given path is matched by the regex expression.
-// recursively test up the tree.
-private bool isPathMatched(string path, Regex!char mask) {
-	path = buildNormalizedPath(path);
-	auto paths = pathSplitter(path);
-
-	string prefix = "";
-	foreach(base; paths) {
-		prefix ~= base;
-		if (!path.matchFirst(mask).empty) {
-			// the given path matches something which we should skip
-			return true;
-		}
-		prefix ~= dirSeparator;
-	}
-	return false;
-}
-
-unittest
-{
-	assert(isPathExcluded("Documents2", ["Documents"]));
-	assert(!isPathExcluded("Documents", ["Documents"]));
-	assert(!isPathExcluded("Documents/a.txt", ["Documents"]));
-	assert(isPathExcluded("Hello/World", ["Hello/John"]));
-	assert(!isPathExcluded(".", ["Documents"]));
-}
+import std.algorithm;
+import std.array;
+import std.file;
+import std.path;
+import std.regex;
+import std.stdio;
+import util;
+
+final class SelectiveSync
+{
+	private string[] paths;
+	private Regex!char mask;
+	private Regex!char dirmask;
+	private bool skipDirStrictMatch = false;
+	private bool skipDotfiles = false;
+	private string[] businessSharedFoldersList;
+	
+	void load(string filepath)
+	{
+		if (exists(filepath)) {
+			paths = File(filepath)
+				.byLine()
+				.map!(a => buildNormalizedPath(a))
+				.filter!(a => a.length > 0)
+				.array;
+		}
+	}
+	
+	// Configure skipDirStrictMatch if function is called
+	// By default, skipDirStrictMatch = false;
+	void setSkipDirStrictMatch()
+	{
+		skipDirStrictMatch = true;
+	}
+
+	void loadSharedFolders(string filepath)
+	{
+		if (exists(filepath)) {
+			businessSharedFoldersList = File(filepath)
+				.byLine()
+				.map!(a => buildNormalizedPath(a))
+				.filter!(a => a.length > 0)
```

Comment:
Could we support comments here, or is this too painful?

Reply:
Potentially ... would have to look into to .. did not think about having comments ....

Reply:
Resolved by https://github.com/abraunegg/onedrive/pull/473/commits/1cee4ee2c5384fc0e774fda9dce5dd48df0818ec